### PR TITLE
Update some dependencies

### LINKED ISSUE #335: Upgrading Prost
Hi, are there any plans to upgrade prost to it's 0.6.1 version? It includes a security fix as in addition to lots of other updates. https://github.com/danburkert/prost/releases/tag/v0.6.1 

I know there are many upstream places that it's used as well in raft's toolchain, if someone could tell me where they all are I can open PRs upgrading all the specific components. I just want to make sure I get them all in one pass.

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

* Update some dependencies
* Upgrade MSRV to 1.39.0 (required by `bytes` v0.5)

Close #335 (waiting for [grpcio-compiler/426](https://github.com/tikv/grpc-rs/pull/426)  and [protobuf-build/34](https://github.com/tikv/protobuf-build/pull/34) to release new version)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ rust:
   - stable
   - nightly
   # Officially the oldest compiler we support.
-  - 1.36.0
+  - 1.39.0
```
Comment (BusyJay) [inline .travis.yml]: Why change this?

Comment (koushiro) [reply .travis.yml]: required by bytes v0.5

**Comment by Fullstop000** [comment]
And why remove `quick-error`?

**Comment by koushiro** [comment]
Why not use std::error::Error + std::fmt::Display? 
The description and cause methods have been deprecated.

**Comment by BusyJay** [comment]
I can see the master of quick_error has resolved the problem. Let's wait for its next release.

And we may switch to other library depends on discussion in tikv/rfcs#38.

**Comment by koushiro** [comment]
Okay, let's set aside the error handling part and wait for the discussion of RFC to have a result that most people agree with.
BTW, I prefer to use the library `thiserror` :smile:

File: Cargo.toml [hunk 2]
```diff
@@ -25,32 +25,32 @@ default-logger = ["slog-stdlog", "slog-envlogger", "slog-term"]
...
-slog-envlogger = { version = "2.1.0", optional = true }
-slog-term = { version = "2.4.0", optional = true }
-quick-error = "1.2.2"
+quick-error = "1"
 raft-proto = { path = "proto", version = "0.6.0-alpha", default-features = false }
 rand = "0.7"
-fxhash = "0.2.1"
-fail = { version = "0.3", optional = true }
-getset = "0.0.7"
+slog = "2"
```
Comment (BusyJay) [inline Cargo.toml]: I suggest to keep all these versions as their original values. For example, it's different between "2" and "2.2". Libraries work with 2.2.0 don't have to work with 2.0.0. Changing it to 2 gives a false assumption that we have tested and guarantee compatible against 2.1 and 2.0.

File: proto/Cargo.toml [hunk 3]
```diff
@@ -15,14 +15,12 @@ build = "build.rs"
 [features]
 default = ["protobuf-codec"]
 protobuf-codec = ["protobuf-build/protobuf-codec"]
-prost-codec = ["prost", "prost-derive", "bytes", "lazy_static", "protobuf-build/prost-codec"]
+prost-codec = ["protobuf-build/prost-codec", "prost", "lazy_static"]
```
Comment (BusyJay) [inline proto/Cargo.toml]: Interesting. Isn't prost requires bytes dependency?

Comment (koushiro) [reply proto/Cargo.toml]: prost requires bytes dependency, but raft does not explicitly use the methods in Buf and BufMut traits

Comment (nrc) [reply proto/Cargo.toml]: Prost 0.6 fixed that requirement, it's only required with 0.5 unless you use Buf/BufMut, etc.

**Pull request merged**